### ISSUE TITLE
Using classnames with css-modules/name-space

### ISSUE THREAD

**Issue by blia** [JedWatson/classnames#56]
Using classnames with css-modules/namespace
================================================

Hi!
I (probably, not only I) like to use rather popular `classnames + css-modules` use case.
The most popular usage if it is not only concatenation but concatenation with truthfulness check.
Disadvantage: with `css-modules`,real class names are usually stored in some object.
Usually it's looking the next way:
```js
import styles from './styles.css'

const className = cx({
  [styles.foo]: someLogic
})
```
And this looks weird.

I suggest some minor changes that will allow to "bind" the context.

As far as I understand that doesn't affect the speed too much. But should allow to make className calculations more comfortable to perform.
Also you can define this feature in a separate `bind.j`s file and wrap into boundCreator.

And to use classnames in a certain way.

```js
const styles = {
  foo: '#foo',
  bar: '#bar',
  baz: '#baz'
}
var cx = require('classnames').bind(styles);
// or 
var cx = require('classnames/bind')(styles);

const className = cx('bar', ['foo'], {baz: true})
// -> #bar #foo #baz

// Or es 6
import styles from './styles.css'
import classnames from 'classnames'

const cx = classnames.bind(styles)
const className = cx('bar', ['foo'], {baz: true})
// -> #bar #foo #baz

// es7
const className = styles::classnames('bar', ['foo'], {baz: true})
// -> #bar #foo #baz
```

**Comment by dcousens** [comment]
I'd be OK with `classnames/bind`

**Comment by JedWatson** [comment]
+1 for `classnames/bind`, will leave a couple of notes in the PR

**Comment by JedWatson** [comment]
Published in `v2.2.0`

**Comment by egauderman** [comment]
For those that are looking for ways to nicely combine classnames and css modules, see this:
https://github.com/itsmepetrov/classnames-loader
